### PR TITLE
Feature/s3 c 2797 bucketinfo update

### PR DESCRIPTION
**Pull request by dora-korpar**
Updates BucketInfo class for bucket notifications

### CODE REVIEW

**Review by naren-rajendran — APPROVED** [review]

**Review by tmacro — APPROVED** [review]

**Review by miniscruff — CHANGES_REQUESTED** [review]
Is it ok to mark this as "Request changes" until the other PR is merged? Considering I had comments on the other one, plus I am hoping to keep this on my own list in github so I do not lose it again.

**Review by jonathan-gramain — APPROVED** [review]
I did not review the code but approving to unblock PR

File: lib/models/BucketInfo.js
Comment (miniscruff) [inline lib/models/BucketInfo.js:412]: When using the constructor the notification configuration is validated, but when setting it is not. Not sure if this discrepency matters but I think they should both be validated.

**Review by miniscruff — APPROVED** [review]
Just a small minor point

File: lib/models/BucketInfo.js
Comment (dora-korpar) [reply lib/models/BucketInfo.js:412]: in cloudserver, the put bucket notification api uses the constructor to validate a config before setting it

Comment (miniscruff) [reply lib/models/BucketInfo.js:412]: I am sorry I do not understand. The put bucket notification uses the constructor to validate a config and then sets the config again after it was validated and stored already in the constructor?

**Comment by dora-korpar** [comment]
@bert-e approve

**Pull request merged**
